hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -42,173 +38,248 @@ object HealthTransformations {
-          )
-        // generic case for infectious disease
-        case (disease, categorical) =>
-          createHealthConditionRow(
-            rawRecord,
-            s"dx_${disease}",
-            infectiousDiseaseCondition,
-            categorical
-          )
-      }
-    } else None
-
-  def mapCongenitalEyeDisorder(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
-    if (rawRecord.getBoolean("hs_congenital_yn") && rawRecord.getBoolean("hs_cg_eye_disorders_yn"))
-      eye_cg_disorders.flatMap {
-        // "other" case
-        case ("other", categorical) =>
-          createHealthConditionRow(
-            rawRecord,
-            "cg_eye_other",
-            eyeDiseaseCondition,
-            categorical,
-            isCongenital = true,
-            conditionOtherDescription = rawRecord.getOptional("hs_cg_eye_other_spec")
-          )
-        // general case
-        case (disorder, categorical) =>
-          createHealthConditionRow(
-            rawRecord,
-            s"cg_eye_${disorder}",
-            eyeDiseaseCondition,
-            categorical,
-            isCongenital = true
-          )
-      }
-    else None
-
-  def mapEyeDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
-    if (rawRecord.getBoolean("hs_dx_eye_yn")) {
-      eye_diseases.flatMap {
-        // "other" case
-        case ("eye_other", categorical) =>
-          createHealthConditionRow(
-            rawRecord,
-            "dx_eye_other",
-            eyeDiseaseCondition,
-            categorical,
-            conditionOtherDescription = rawRecord.getOptional("hs_dx_eye_other_spec")
-          )
-        case ("blind", categorical) => {
-          val isCauseKnown = rawRecord.getBoolean("hs_dx_eye_cause_yn")
-          val conditionCause =
-            if (isCauseKnown) rawRecord.getOptionalNumber("hs_dx_eye_cause") else None
-          createHealthConditionRow(
-            rawRecord,
-            "dx_blind",
-            eyeDiseaseCondition,
-            categorical,
-            conditionCause = conditionCause,
-            conditionCauseOtherDescription =
-              if (isCauseKnown && (conditionCause.getOrElse(None) == 98))
-                rawRecord.getOptional("hs_dx_eye_cause_other")
-              else None
-          )
-        }
-        case (disease, categorical) =>
-          createHealthConditionRow(rawRecord, s"dx_${disease}", eyeDiseaseCondition, categorical)
-      }
-    } else None
+  def processHealthCondition(
+    rawRecord: RawRecord,
+    healthCondition: HealthCondition
+  ): Iterable[HlesHealthCondition] = {
+    val congenital: Option[HlesHealthCondition] = healthCondition.congenital.flatMap {
+      case (target, dependent) =>
+        // if congenital, then check hs_congenital_yn && healthCondition.congenital._2
+        if (rawRecord.getBoolean("hs_congenital_yn") &&
+            rawRecord.getBoolean(s"hs_${dependent}_yn"))
+          // congenital other
+          if (healthCondition.isOther)
```

Comment:
Style nit: can you add brackets around the `if` and `else` arms across the board? And make sure the bodies are pushed to newlines:
```scala
if (thing) {
 // foo
} else if (otherThing) {
  // bar
} else {
 // baz
}
```